rfcs/0010-transition-frontier-bootstrap.md

```diff
@@ -0,0 +1,67 @@
+## Summary
+[summary]: #summary
+
+The goal of this RFC is to create a simple process for bootstrapping a new node or a node who has been inactive for a long time to the network. This is done by asking a peer for its root external transition and a proof of it along with the staged ledger scan state corresponding to the transition. Bootstrap allows a node to forgoe its entire transition frontier and sync with the network by syncing just with the network's root.
+
+## Motivation
+[motivation]: #motivation
+
+With the derivations constructed by Oroborous Praos/Genesis, we can assume that finality is achieved at the a node's root external transition. Thus, there is only one uniquly valid `snarked_ledger` and `staged_ledger` that a node has to bootstrap to. In this RFC, we discuss how we can achieve this efficiently. 
+
+## Detailed design
+[detailed-design]: #detailed-design
+
+### Constants
+
+Thoughout the design, we will describe terms in relation to a constant `K`. The constant `K` is a natural number representing the transitions before finality can be assumed. As per Ouroboros proof of stake, `K = unforkable_slot_count = 2160`. The maximum length of a branch in the transition frontier is `K`, as the transition frontier root represents the point of finality of the network.
+
+### Activation Conditions
+
+A node would want to bootstrap when it sees that is not synced to the network compared to its peers. It does this by comparing the length of its root's consensus state and the length of the consensus state of the transition that it currently sees from its peers, `seen_transition`. A node can see if the length of its peer's root consensus state is at least `seen_transition.length - K`. If this value is greater than  the node's current root length, then the node is out of sync with the network compared to its peer. This peer's root may exist in the node's frontier, so it would not be too out of synced and the node could run a catchup job to be in synced with the network (See [ledger_catchup.ml](src/lib/ledger_catchup/ledger_catchup.ml) for more details). It would very out of sync when this does not occur. As a result, a node would want to bootstrap when `seen_transition.length - root.length > 2k`.
```

Comment:
When reading this I found it hard to keep straight node vs peer. It doesn't help that multiple meanings of "node" are in play. Maybe we can rewrite this with names for the protocol participants: Alice is trying to bootstrap, and she is getting the proofs she needs from Bob.